src/sqltest/services/connectionManagement/connectionManagementService.test.ts

```diff
@@ -0,0 +1,170 @@
+			defaultValue: undefined,
+			isIdentity: true,
+			isRequired: true,
+			specialValueType: ConnectionOptionSpecialType.userName,
+			valueType: ServiceOptionType.string
+		},
+		{
+			name: 'authenticationType',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: true,
+			isRequired: true,
+			specialValueType: ConnectionOptionSpecialType.authType,
+			valueType: ServiceOptionType.string
+		},
+		{
+			name: 'password',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: true,
+			isRequired: true,
+			specialValueType: ConnectionOptionSpecialType.password,
+			valueType: ServiceOptionType.string
+		},
+		{
+			name: 'encrypt',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: false,
+			isRequired: false,
+			specialValueType: undefined,
+			valueType: ServiceOptionType.string
+		}
+	]
+};
+
+const capabilitiesService = new CapabilitiesTestService();
+capabilitiesService.capabilities['MSSQL'] = { connection: sqlProvider };
+
+suite('ConnectionManagementService Tests:', () => {
+	test('getConnections test', () => {
+		const connectionStatusManagerMock = TypeMoq.Mock.ofType(ConnectionStatusManager, TypeMoq.MockBehavior.Loose);
+		const connectionStoreMock = TypeMoq.Mock.ofType(ConnectionStore, TypeMoq.MockBehavior.Loose, new TestStorageService());
+
+		connectionStatusManagerMock.setup(x => x.getActiveConnectionProfiles(undefined)).returns(() => {
+			return [createConnectionProfile('1'), createConnectionProfile('2')];
+		});
+		connectionStoreMock.setup(x => x.getRecentlyUsedConnections(undefined)).returns(() => {
+			return [createConnectionProfile('1'), createConnectionProfile('3')];
+		});
+
+		const group1 = createConnectionGroup('group1');
+		const group2 = createConnectionGroup('group2');
+		group1.connections = [createConnectionProfile('1'), createConnectionProfile('4')];
+		group1.children = [group2];
+		group2.connections = [createConnectionProfile('5'), createConnectionProfile('6')];
+		connectionStoreMock.setup(x => x.getConnectionProfileGroups(TypeMoq.It.isAny(), undefined)).returns(() => {
+			return [group1];
+		});
+		const connectionManagementService = new ConnectionManagementService(connectionStoreMock.object, connectionStatusManagerMock.object, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined);
+
+		// dupe connections have been seeded the numbers below already reflected the de-duped results
+
+		// no parameter - default to false
+		let connections = connectionManagementService.getConnections();
+		assert.equal(connections.length, 6, 'no known connections');
+
```

Comment:
1. The assert message isn't very useful - how is this going to help me know what is expected and what was wrong? If connections returns 5 connections this will still fail but say "no known connections" which isn't very useful
2. It's really helpful to always print out what you DID get for failures like this. In this case print out the stringified version of connections
3. It'd be even better to use something like assert.deepEqual to compare the actual contents of the arrays. This would avoid the issue right now that if for some reason we were returning duplicate connections the test still might pass

Reply:
copy paste issue, i have fixed the error messages and also check the connection id.